### DISCUSSION TITLE
Sotodlib Wrap Function Channel Matching

### DISCUSSION THREAD

**Discussion by samdayweiss** [simonsobs/sotodlib#541] [Q&A]
Hi all,

Does the wrap function guarantee that the channels (in smurf band, sub band)  in the original axis manager (from 'load_file') are matched up to the same channels in arrays that are wrapped afterwards (e.g. iv data or biasstep data)? 

For example, in the below, `aman[ufm]` is the axis manager object consisting of the signal,timestamps, and also detector map info after running `g3utils.add_detmap_info`

`bias_step_ana = np.load(bstep_dict[ufm], allow_pickle=True).item()`
`bsman = sdl.core.AxisManager(aman_dict[ufm].dets, aman_dict[ufm].bias_lines)`
`bsman.wrap('bgmap', bias_step_ana['bgmap'], [(0, 'dets')]`
`bsman.wrap('si', bias_step_ana['Si'], [(0, 'dets')])`
`bsman.wrap('R0', bias_step_ana['R0'], [(0, 'dets')])`
`aman_dict[ufm].wrap('biasstep', bsman)`

My question is whether the responsivity, `aman.biasstep['Si']`, for example, is guaranteed to be indexed by the same channels as the original tod in the axis manager object? As far as I can tell it does, but I can't find anywhere in the docs that explains that this is the case/how it is done.

I should also add: does the  `g3utils.add_detmap_info` guarantee proper indexing?

Thanks!

**Comment by msilvafe** [comment]
Hi Sam, yes they are aligned! My understand is that this is one of the purpose of axis manager data types to create 'axes' with are named so that when you wrap new things and specify along which axis this are wrapped it will align the indexing without worrying about transposing or counting indices of high dimensional arrays. And if you reduce one axis it'll apply that to all arrays aligned along that axis so they're all still indexed properly. I think this should be explained in the docs on axis managers and/or in the pwg-tutorials notebooks!
